src/compiler/types.ts

```diff
@@ -293,6 +293,8 @@ module ts {
         Const =             0x00001000,  // Variable declaration
         OctalLiteral =      0x00002000,
 
+        SimpleArrowFunction =      0x00004000,  // Arrow function without parenthesized parameters

```

Comment:
You could do the trick where you compare the beginning of the function expression to the beginning of the parameter. So if it has one parameter, whose pos coincides with the pos of the function expression, it's a simple arrow